RadixWallet/Clients/ROLAClient/ROLAClient+Live.swift

```diff
@@ -10,35 +10,38 @@ extension ROLAClient {
 			let entity = request.entity
 			let newPublicKey = request.newPublicKey
 
-			let entityAddress: Address = switch entity {
-			case let .account(account):
-				account.address.asGeneral
-			case let .persona(persona):
-				persona.address.asGeneral
-			}
-			let metadata = try await onLedgerEntitiesClient.getEntity(entityAddress, metadataKeys: [.ownerKeys]).genericComponent?.metadata
+			let entityAddress: AddressOfAccountOrPersona = {
+				switch entity {
+				case let .account(account):
+					assert(account.networkID == request.entity.networkID)
+					return AddressOfAccountOrPersona.account(account.address)
+				case let .persona(persona):
+					assert(persona.networkID == request.entity.networkID)
+					return AddressOfAccountOrPersona.identity(persona.address)
+				}
+			}()
+
+			let metadata = try await onLedgerEntitiesClient.getEntity(entityAddress.embed(), metadataKeys: [.ownerKeys]).genericComponent?.metadata
 			var ownerKeyHashes = try metadata?.ownerKeyHashes() ?? []
 
 			let transactionSigningKeyHash: PublicKeyHash = switch entity.securityState {
 			case let .unsecured(control):
-				try .init(hashing: control.transactionSigning.publicKey)
+				.init(hashing: control.transactionSigning.publicKey.intoSargon())
```

Comment:
We have keys in iOS wallet AND Sargon now - so need to map some keys into Sargon ones until we have migrated over to using Profile in Sargon